daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/Parser.scala

```diff
@@ -281,6 +293,10 @@ class ChoiceParser(ctxt: RuntimeData, val childParsers: Vector[Parser])
       }
 
       pstate.popDiscriminator
+
+      if (discriminatorFailed)
+        pstate.setDiscriminator(true)
+
```

Comment:
Is this setting the correct discriminator? I was originally thinking that we would need to push a new discriminator somewhere, and then check that result someehwere else to know when to bail on the unordered sequence. But maybe that's not necessary and this is correct? This could maybe use a comment to explain where the discrimnator that this is setting gets pushed onto the stack?

Reply:
The way I viewed the discriminator stack (which may be incorrect) is that the discriminator we are actually interested in is on the unordered sequence, but because we are implementing uoseq's with a ChoiceParser, that discriminator essentially gets pushed one level down to the ChoiceParser.  By setting the discriminator here, we are essentially just lifting it out of the ChoiceParser and back into the Unordered Sequence.

I added comments to the part of the code where discriminatorFailed gets set to true, but perhaps I should add something here too.

Reply:
Add comment to this setDiscriminator call pointing out that the inner choice discriminator is now off the stack, so this is setting the discriminator of the outer unordered sequence loop insuring that while we have two physical discriminators (outer for the loop, inner for the choice), that this is making it behave as if there was only a single discriminator.

Reply:
Done